cmake/scripts/common/ArchSetup.cmake

```diff
@@ -157,3 +157,9 @@ if(CMAKE_BUILD_TYPE STREQUAL "Debug")
   add_options (ALL_LANGUAGES DEBUG "-g" "-D_DEBUG" "-Wall")
 endif()
 
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_options(CXX "DEBUG;RELEASE" "-Wno-overloaded-virtual")
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" AND (CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 7 OR CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 7))
```

Comment:
I couldn't find the exact version for clang where it got first introduced, will add it and also check if it's needed for release builds.